libtcmu_log.c

```diff
@@ -697,6 +695,7 @@ int tcmu_setup_log(char *log_dir)
 	if (ret < 0)
 		tcmu_err("create file output error \n");
 
+	tcmu_logbuf = logbuf;
 	ret = pthread_create(&logbuf->thread_id, NULL, log_thread_start,
 			     logbuf);
 	if (ret) {
```

Comment:
Nice debugging.

Could you just add a:

tcmu_logbuf = NULL;

in this error path. If for debugging the caller does something like:

if (tcmu_setup_log())
     tcmu_err("setup failed");

then it will not crash.